packages/app-store/googlecalendar/api/callback.ts

```diff
@@ -50,6 +50,34 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
 
+  // Fetch the user's email from the Google OAuth2 userinfo endpoint using the access token in 'key'.
+  const userEmail = await fetch("https://www.googleapis.com/oauth2/v3/userinfo", {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${key?.access_token}`,
+    },
+  })
+    .then((res) => res.json())
+    .then((res) => res.email); // Extract the user's email from the response.
+
+  // Upsert (Insert or Update) a record of primary calendar in the 'selectedCalendar' table in the Prisma database.
+  await prisma.selectedCalendar.upsert({
+    where: {
+      userId_integration_externalId: {
+        userId: req.session.user.id,
+        integration: credentials?.type, // Match the credential type.
+        externalId: userEmail, // Match the user's email.
+      },
+    },
+    create: {
+      userId: req.session.user.id,
+      integration: credentials?.type, // Set the integration type from the credential.
+      externalId: userEmail, // Set the external ID to the user's email.
+      credentialId: credentials?.id, // Associate the credential with this record.
+    },
+    update: {},
```

Comment:
creating a record in database to turn the toggle on by default when google calendar app is installed.
userId: taken from session
integration: taken from credentials
externalId: taken from userEmail
credentialId: taken from credentials

Reply:
> `question`: Why is this operation being done as an upsert when there's no actual update happening? If in fact, this is only called when the integration is enabled, an insert would suffice.

[f0863a1](https://github.com/calcom/cal.com/pull/11639/commits/f0863a129bc9e7d76593db1d57c865b34f86d444)
I've updated the code to use the 'insert' method since there's no actual update happening.

Reply:
Thank you!